Intel_SighPP_CMake/Intel_SighPP_CMake/src/priority_lib/smart_priority.hpp

```diff
@@ -1,76 +1,155 @@
 #pragma once
 
 #include "priority_module.hpp"
+#include "priority_clock.hpp"
 #include <spdlog/spdlog.h>
+#include "../classification_result.hpp"
 #include <string>
+#include <map>
 
 class smart_priority : public priority_module
 {
 protected:
 	int const size_w = 640;
 	int const size_h = 480;
+	int const cooldown = 5000;
+	int const cooldown_shortening = 1000;
+	int const cooldown_emergency = 2500;
+	double const walking_speed = 1.4;
 	int centre_line = 0;
 	int out_left = 0;
 	int out_right = 0;
 	int in_left = 0;
 	int in_right = 0;
 	int above = 0;
-	int bottom  = 0;
+	int bottom = 0;
 	int mid_w = 0;
+	priority_clock pc;
+	Priority min = Priority::UNDEFINED;
 
 public:
 
+	/// <summary>
+	///Object Creation. Not to be inherited furhter.
+	/// Initialises location markers based on ratio
+	/// </summary>
+	/// <param name="name"></param>
+	/// <returns></returns>
 	smart_priority(std::string name = "default")
 		: priority_module(&name) {
 		determine_location_markers();
 	}
-
+	/// <summary>
+	/// Determines location markers for objects within the screen. 
+	/// </summary>
 	void determine_location_markers();
+
+
+	//________data and message control________: 
+
+	/// <summary>
+	/// Based on rectangle of object determin location
+	/// </summary>
+	/// <param name="item"></param>
 	void msg_add_location(ClassificationItem& item);
+
+	/// <summary>
+	/// Add name of object to message for interface
+	/// </summary>
+	/// <param name="item"></param>
 	void msg_add_name(ClassificationItem& item);
+
+	/// <summary>
+	/// Add distance of object to message for interface
+	/// </summary>
+	/// <param name="item"></param>
 	void msg_add_distance(ClassificationItem& item);
+
+	/// <summary>
+	/// Calculate speed until impact with user. Note does not yet take location into account.
+	/// </summary>
+	/// <param name="item"></param>
+	double time_until_colision(ClassificationItem& item);
+
+	//__________Tracking checking rules______________
+
+	/// <summary>
+	/// Tracks the main cooldown of each object. If item is out of cooldown then it 
+	/// is approved to run through the priority rules. If not it is currently 
+	/// set to low prio.
+	///
+	/// Note: Items that are tracked but not currently seen are also set to low prio. This could be improved after report.
```

Comment:
Maybe better to include the last part as a // TODO, instead of a documented note 😄